addons/sourcemod/scripting/spec-lister.sp

```diff
@@ -23,82 +16,55 @@ public Plugin:myinfo =
 	url = "http://forums.alliedmods.net/showthread.php?t=95474"
 }
 
-
- public OnPluginStart()
+public OnPluginStart()
 {
-	HookEvent("player_team",Event_PlayerChangeTeam);
-	RegConsoleCmd("hear", Panel_hear);
-
+	RegConsoleCmd("hear", PanelCommand);
 }
-public PanelHandler1(Handle:menu, MenuAction:action, param1, param2)
+
+public PanelHandler(Handle:menu, MenuAction:action, client, selectedValue)
 {
 	if (action == MenuAction_Select)
 	{
-		PrintToConsole(param1, "You selected item: %d", param2)
-		if(param2==1)
-			{
-			SetClientListeningFlags(param1, VOICE_LISTENALL);
-			PrintToChat(param1,"\x04[listen]\x03enable" );
-			}
+		PrintToConsole(client, "You selected item: %d", selectedValue)
+
+		if(selectedValue == 1)
+		{
+			SetClientListeningFlags(client, VOICE_LISTENALL);
+			PrintToChat(client,"\x04[Ouvir] \x03Habilitado");
+		}
 		else
-			{
-			SetClientListeningFlags(param1, VOICE_NORMAL);
-			PrintToChat(param1,"\x04[listen]\x03disable" );
-			}
-		
-	} else if (action == MenuAction_Cancel) {
-		PrintToServer("Client %d's menu was cancelled.  Reason: %d", param1, param2);
-	}
+		{
+			SetClientListeningFlags(client, VOICE_NORMAL);
+			PrintToChat(client,"\x04[Ouvir] \x03Desabilitado");
+		}
+	} 
 }
 
-
-public Action:Panel_hear(client,args)
+public Action:PanelCommand(client,args)
 {
-	if(GetClientTeam(client)!=TEAM_SPEC)
+	if(GetClientTeam(client) != TEAM_SPEC)
 		return Plugin_Handled;
+
 	new Handle:panel = CreatePanel();
-	SetPanelTitle(panel, "Enable listen mode ?");
-	DrawPanelItem(panel, "yes");
-	DrawPanelItem(panel, "no");
+
+	SetPanelTitle(panel, "Deseja ouvir os jogadores?");
+	DrawPanelItem(panel, "Sim");
+	DrawPanelItem(panel, "Não");
  
-	SendPanelToClient(panel, client, PanelHandler1, 20);
+	SendPanelToClient(panel, client, PanelHandler, 20);
  
 	CloseHandle(panel);
  
 	return Plugin_Handled;
-
 }
 
 public OnClientPutInServer(client)
 {
-CreateTimer(40.0, TimerAnnounce, client);
+	CreateTimer(40.0, TimerAnnounce, client);
 }
 
 public Action:TimerAnnounce(Handle:timer, any:client)
 {
-if (IsClientInGame(client))
-	PrintToChat(client,"\x04[listen]enable for spector only�G\03!hear" );
-}
-
-public Event_PlayerChangeTeam(Handle:event, const String:name[], bool:dontBroadcast)
-{
-	new userID = GetClientOfUserId(GetEventInt(event, "userid"));
-	new userTeam = GetEventInt(event, "team");
-	if(userID==0)
-		return ;
-
-	//PrintToChat(userID,"\x02X02 \x03X03 \x04X04 \x05X05 ");\\ \x02:color:default \x03:lightgreen \x04:orange \x05:darkgreen
-	
-	if(userTeam==TEAM_SPEC)
```

Comment:
Veja que aqui ele verifica se o jogador foi pra spec ou não, se for != SPEC ele volta o audio do jogador ao normal
SetClientListeningFlags(userID, VOICE_NORMAL);